### ISSUE TITLE
useCopyToClipboard is not working in chat share dialog in Safari

### ISSUE THREAD

**Issue by a-mkh** [vercel/ai-chatbot#283]
`Unhandled Promise Rejection: NotAllowedError: The request is not allowed by the user agent or the platform in the current context, possibly because the user denied permission.`

Both desktop and mobile safari.

Possible workaround is to copy url to clipboard prior to await for server action.

**Comment by BaileySimrell** [comment]
Did you ever develop a workaround for this?

**Comment by a-mkh** [comment]
@BaileySimrell Yeah, this is a quick and dirty fix in share chat dialog as an example - copying to clipboard should be called prior to running a server action, which breaks the way component was built.
`
          <Button
            disabled={isSharePending}
            onClick={() => {
              // @ts-ignore
              startShareTransition(async () => {
                const url = new URL(window.location.href)
                url.pathname = `/share/${chat.id}`
                copyToClipboard(url.toString())

                const result = await shareChat(chat.id)

                if (result && 'error' in result) {
                  toast.error(result.error)
                  return
                }
                if (navigator.userAgent.indexOf("Safari") == -1) {
                  copyShareLink(result)
                }
              })
            }}
          >
`

**Comment by BaileySimrell** [comment]
> @BaileySimrell Yeah, this is a quick and dirty fix in share chat dialog as an example - copying to clipboard should be called prior to running a server action, which breaks the way component was built.
> 
> ```
>           <Button
>             disabled={isSharePending}
>             onClick={() => {
>               // @ts-ignore
>               startShareTransition(async () => {
>                 const url = new URL(window.location.href)
>                 url.pathname = `/share/${chat.id}`
>                 copyToClipboard(url.toString())
> 
>                 const result = await shareChat(chat.id)
> 
>                 if (result && 'error' in result) {
>                   toast.error(result.error)
>                   return
>                 }
>                 if (navigator.userAgent.indexOf("Safari") == -1) {
>                   copyShareLink(result)
>                 }
>               })
>             }}
>           >
> ```

@a-mkh Amazing man, thank you so much! That worked for me.

I added an else statement after the `if (result && 'error' in result)` to send the toast success state because the old promised based toast method stopped working.

**Comment by athrael-soju** [comment]
Does this fix the issue of the wrong number of messages being shown at the share chat dialog?
